epoxy-processor/src/main/java/com/airbnb/epoxy/EpoxyProcessor.java

```diff
@@ -245,12 +258,29 @@ private void updateClassesForInheritance(
 
       for (Entry<TypeElement, ClassToGenerateInfo> otherEntry : otherClasses.entrySet()) {
         if (isSubtype(entry.getKey().asType(), otherEntry.getKey().asType())) {
-          entry.getValue().addAttributes(otherEntry.getValue().getAttributeInfo());
+          if (belongToTheSamePackage(entry.getKey(), otherEntry.getKey())) {
+            entry.getValue().addAttributes(otherEntry.getValue().getAttributeInfo());
+          } else {
+            for (AttributeInfo attribute : otherEntry.getValue().getAttributeInfo()) {
```

Comment:
A comment here about why we do this would probably be helpful, just going back over this code that I haven't looked at in a few months I'm already having a hard time remembering what it does :P

If you prefer you can leave it for now and I can comments later after merging